Platform/obc/eps.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file eps.c
+ * @author Andrew Rooney
+ * @date 2020-12-28
+ */
+
+/*
+ * This code needs to:
+ *  - send requests to the EPS over the network as per the NanoAvionics specs
+ *  - provide an API around those calls for the on-board applications to use
+ */
+
+#include <FreeRTOS.h>
+#include <task.h>
+#include <csp/csp.h>
+
+#include "eps.h"
+#include "system.h"
+
+SAT_returnState eps_get_instantaneous_telemetery(eps_instantaneous_telemetry_t *telembuf) {
+    uint8_t cmd = 0;
+    if (telembuf == NULL) {
+        return SATR_ERROR;
+    }
+    csp_transaction_w_opts(CSP_PRIO_LOW, (TC_TM_app_id) EPS_APP_ID,

```

Comment:
Also, won't the obc itself have a use for eps data? I mean starting a task once some value changes in eps data when there's no TM request.